### PR TITLE
Added tag and provider expansion for the web interface

### PR DESCRIPTION
**Pull request by zquestz**
This just adds tag and provider expansion for the web front-end. Now we can do the following queries:

Search duckduckgo for "foo"
```
https://<domain>/search?q=foo&provider=duck
```

Search yahoo and the tag music for radiohead
```
https://<domain>/search?q=radiohead&provider=yah&tag=mu
```

@akb @tw4dl @mbhinder @lilirui @KeizerDev

### CODE REVIEW

**Comment by mbhinder** [comment]
LGTM

File: server/search.go [hunk 1]
```diff
@@ -86,18 +91,14 @@ func executeSearch(providerList []providers.Provider, query, locale string) []st
...
-func tagProvidersNotFound(tag string, w http.ResponseWriter, r *http.Request) {
-	http.Error(w, fmt.Sprintf("No providers matched tag %q.", tag), 400)
-}
-
 func queryNotFound(w http.ResponseWriter, r *http.Request) {
 	http.Error(w, fmt.Sprintf("A search query is required."), 400)
 }
 
+func expandNotValid(err error, w http.ResponseWriter, r *http.Request) {
+	http.Error(w, err.Error(), 400)
```
Comment (lilirui) [inline server/search.go]: Nit: I like replace 400 with http.StatusBadRequest

**Comment by lilirui** [comment]
LGTM

File: server/search.go (hunk 1 above)
Comment (zquestz) [inline server/search.go]: Agreed, will fix.

**Pull request merged**